data/split_step_generator.py

```diff
@@ -28,63 +29,151 @@ def __init__(self,
 
-    def prepare_data(self, a=None):
-        if a is None:
-            a = 2*np.random.randint(1,3, size=self.seq_len) - 3
-        else:
-            self.seq_len = len(a)
-        
-        z = torch.linspace(0, self.z_end, self.num_z_points)
+        Returns
+        -------
+        t : TYPE: torch.float32 tensor of shape [dim_t]
+            DESCRIPTION: Time points. See prepare_data description.
+        z : TYPE: torch.float32 tensor of shape [dim_z]
+            DESCRIPTION: Points in space. See prepare_data description.
+        u : TYPE: torch.complex128 tensor of shape [batch_size, dim_z, dim_t].
+            DESCRIPTION: Output of the split-step solution.
+        '''
+        z = torch.linspace(0, L, self.dim_z)
         
-        tmax = (self.seq_len + 1) * self.pulse_width
-        tMax = tmax + 5*np.sqrt(2*(1 + self.z_end**2))
+        tMax = self.t_end + 5*np.sqrt(2*(1 + L**2))
         tMin = -tMax
         
-        dt = (tMax - tMin) / self.num_t_points
-        t = torch.linspace(tMin, tMax-dt, self.num_t_points)
+        dt = (tMax - tMin) / self.dim_t
+        t = torch.linspace(tMin, tMax-dt, self.dim_t)
         
         # prepare frequencies
         dw = 2*np.pi/(tMax - tMin)
-        w = dw*torch.cat((torch.arange(0, self.num_t_points/2+1),
-                          torch.arange(-self.num_t_points/2+1, 0)))
+        w = dw*torch.cat((torch.arange(0, self.dim_t/2+1),
+                          torch.arange(-self.dim_t/2+1, 0)))
         
         # prepare linear propagator
-        LP = torch.exp(-1j*self.dispersion*self.dz/2*w**2)
+        LP = torch.exp(-1j*d*self.dz/2*w**2)
         
         # Set initial condition
-        u = torch.zeros(self.num_z_points, self.num_t_points, dtype=torch.complex128)
+        u = torch.zeros(self.batch_size, self.dim_z, self.dim_t, dtype=torch.complex128)
         
-        buf = self.Etanal(t, torch.tensor(0), a, self.pulse_width)
-        u[0,:] = buf
+        buf = self.Etanal(t, torch.tensor(0), a, T)
+        u[:,0,:] = buf
         
         n = 0
         # Numerical integration (split-step)
-        for i in range(1, int(self.z_end / self.dz) + 1):
+        for i in range(1, int(L / self.dz) + 1):
             buf = ifft(LP * fft(buf))
-            buf = buf*torch.exp(1j*self.nonlinearity*self.dz*buf.abs()**2)
+            buf = buf*torch.exp(1j*c*self.dz*buf.abs()**2)
             buf = ifft(LP*fft(buf))
 
             if i % self.z_stride == 0:
                 n += 1
-                u[n,:] = buf
+                u[:,n,:] = buf
         
         # Dispersion compensation procedure (back propagation D**(-1))
         if self.dispersion_compensate:
             zw = torch.mm(z.view(z.shape[-1], 1), w.view(1, w.shape[-1])**2)
-            u = ifft(torch.exp(1j*self.dispersion*zw)*fft(u))
+            u = ifft(torch.exp(1j*d*zw)*fft(u))
         
         return t, z, u
 
+    def prepare_data(self, two_dim_data=False, a=None):
```

Comment:
`prepare_data` is a method from the base class, I think you can't add arguments to it. Need to move them into a constructor, or use hardcoded.

Reply:
I'll try to do it myself